### PR TITLE
Regions refactoring

### PR DESCRIPTION
**Pull request by EduardoCHenrique**

### CODE REVIEW

**Comment by f3liperamos** [comment]
:+1:

**Comment by albertossilva** [comment]
Eduardo, you don't have to commit the `dist` folder.

File: package.json [hunk 1]
```diff
@@ -20,20 +20,23 @@
     "postinstall": "babel ./src -d lib"
   },
   "devDependencies": {
+    "Faker": "^0.7.2",
```
Comment (albertossilva) [inline package.json]: I think you should write in lowercase here.

File: spec/Regions.spec.jsx [hunk 2]
```diff
@@ -59,4 +91,77 @@ describe( 'Regions module', function () {
...
+    spyOn(Regions.prototype, "refresh");
+    //When
+    var fetchCheck = Regions.fetch( regions, routesMock );
+    //Then
+    expect(Regions.prototype.cachedView).not.toBeUndefined();
+    expect(Regions.prototype.refresh).toHaveBeenCalled();
+  });
+
+  describe(' - refresh - ', function(){
+    const ramdomParamOne = Faker.lorem.words()[0];
```
Comment (albertossilva) [inline spec/Regions.spec.jsx]: Ramdom is written wrong. And... it's is not the param it's the route.

File: spec/Regions.spec.jsx [hunk 3]
```diff
@@ -59,4 +91,77 @@ describe( 'Regions module', function () {
...
+
+  it('should return create a regions instance and call refresh method', function() {
+    //Given
+        const region = (<Region title="main" key="main" main routeFragment="@/aux/*"/>),
+          regions = (<Regions>{[region]}</Regions>);
+
+    Regions.prototype.cachedView = undefined;
+    spyOn(Regions.prototype, "refresh");
+    //When
+    var fetchCheck = Regions.fetch( regions, routesMock );
```
Comment (albertossilva) [inline spec/Regions.spec.jsx]: I think you should use const or let here.

**Comment by albertossilva** [comment]
:+1:

**Pull request merged**